libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1689,6 +1835,7 @@ class MapboxNavigation @VisibleForTesting internal constructor(
      * This function returns `null` for primary route in the current session.
      */
     fun getAlternativeMetadataFor(navigationRoute: NavigationRoute): AlternativeRouteMetadata? {
+        //TODO: make it thread safe?
         return routeAlternativesController.getMetadataFor(navigationRoute)
```

Comment:
`routeAlternativesController` is created on the SDK thread, so yes, it would make sense to make it thread-safe.